searchlib/src/vespa/searchlib/attribute/multistringattribute.hpp

```diff
@@ -62,47 +62,47 @@ class EnumAccessor {
 }
 
 template <typename B, typename M>
-bool
-MultiValueStringAttributeT<B, M>::StringSetImplSearchContext::onCmp(DocId doc, int32_t & weight) const
+int32_t
+MultiValueStringAttributeT<B, M>::StringSetImplSearchContext::onFind(DocId doc, int32_t elemId, int32_t &weight) const
 {
     StringAttribute::StringSearchContext::CollectWeight collector;
-    return this->collectWeight(doc, weight, collector);
+    return this->collectWeight(doc, elemId, weight, collector);
 }
 
 template <typename B, typename M>
-bool
-MultiValueStringAttributeT<B, M>::StringArrayImplSearchContext::onCmp(DocId doc, int32_t & weight) const
+int32_t
+MultiValueStringAttributeT<B, M>::StringArrayImplSearchContext::onFind(DocId doc, int32_t elemId, int32_t &weight) const
 {
     StringAttribute::StringSearchContext::CollectHitCount collector;
-    return this->collectWeight(doc, weight, collector);
+    return this->collectWeight(doc, elemId, weight, collector);
 }
 
 template <typename B, typename M>
 template <typename Collector>
-bool
-MultiValueStringAttributeT<B, M>::StringImplSearchContext::collectWeight(DocId doc, int32_t & weight, Collector & collector) const
+int32_t
+MultiValueStringAttributeT<B, M>::StringImplSearchContext::collectWeight(DocId doc, int32_t elemId, int32_t & weight, Collector & collector) const
```

Comment:
this sounds like it should aggregate and not iterate...

Reply:
Renamed to findNextXX